docs/src/toml-files.md

```diff
@@ -0,0 +1,224 @@
+* After 1.0.0 only make breaking changes when incrementing the major version.
+* After 1.0.0 no new public API should be added without incrementing the
+  minor version. This includes, in particular, new types, functions, methods and
+  method overloads, from e.g. `Base` or other packages.
+
+
+### The `[deps]` section
+
+All dependencies of the package/project are listed in the `[deps]` section.
+Each dependency is listed as a name-uuid pair, for example:
+
+```toml
+[deps]
+Example = "7876af07-990d-54b4-ab0e-23690620f79a"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+```
+
+Typically it is not needed to manually add entries to the `[deps]` section,
+this is instead handled by Pkg operations such as `add`.
+
+
+### The `[compat]` section
+
+Compatibility constraints for the dependencies listed under `[deps]`
+can be listed in the `[compat]` section.
+Example:
+
+```toml
+[deps]
+Example = "7876af07-990d-54b4-ab0e-23690620f79a"
+
+[compat]
+Example = "1.2"
+```
+
+It is also possible to list constraints on `julia` itself, although `julia` is not
+listed as a dependency.
+
+```toml
+[compat]
+julia = "1.1"
+```
+
+The [Compatibility](@ref) section describes the different possible
+compatibility constraints in detail.
+
+
+## `Manifest.toml`
+
+The manifest file is an absolute record of the state of the packages in the environment.
+It includes exact information about (direct and indirect) dependencies of the project,
+and given a `Project.toml` + `Manifest.toml` pair it is possible to instantiate the
+exact same package environment, which is very useful for reproducibility,
+see [`Pkg.instantiate`](@ref).
+
+!!! note
+    The `Manifest.toml` file is generated by Pkg and in general this file
+    should never be modified manually.
+
+
+### `Manifest.toml` entries
+
+Each dependency has its own section in the manifest file, and its content varies depending
+on how the dependency was added, see the examples below.
+Every dependency section includes a combination of the following entries:
+
+* `uuid`: the [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier)
+  for the dependency, for example `uuid = "7876af07-990d-54b4-ab0e-23690620f79a"`.
+* `deps`: a vector listing the dependencies of the dependency, for example
+  `deps = ["Example", "JSON"]`.
+* `version`: a version number, for example `version = "1.2.6"`.
+* `path`: a file path to the source code, for example `path = /home/user/Example`.
+* `repo-url`: a URL to the repository where the source code was found,
+  for example `repo-url = "https://github.com/JuliaLang/Example.jl.git"`.
+* `repo-rev`: a git revision, for example a branch `repo-rev = "master"`
+  or a commit `repo-rev = "66607a62a83cb07ab18c0b35c038fcd62987c9b1"`.
+* `git-tree-sha1`: a content hash of the source tree, for example
+  `git-tree-sha1 = "ca3820cc4e66f473467d912c4b2b3ae5dc968444"`.
+
+Note, however, that `uuid`, `deps` and `version` are present for all dependency sections.
```

Comment:
Only `uuid` is required

Reply:
To elaborate: stdlibs don't really have a `version` and nodes without dependencies of course wont have a `deps` field :+1: